--- mussel/mussel.sh.origII	2024-03-23 17:49:13.598243760 -0500
+++ mussel/mussel.sh	2024-03-23 17:47:41.569247609 -0500
@@ -214,6 +214,7 @@
 
 MPREFIX="/cgnutools"
 MSYSROOT="/cgnutools"
+TVEND="pc"
 
 # ----- mussel Log File ---- #
 MLOG="$CURDIR/log.txt"
@@ -234,7 +235,7 @@
       MARCH=$XARCH
       XGCCARGS="--with-arch=armv8-a --with-abi=lp64 --enable-fix-cortex-a53-835769 --enable-fix-cortex-a53-843419"
       XPURE64=$XARCH
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     armv4 | armv4t )
       XARCH=armv4t
@@ -242,7 +243,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-arch=armv4t --with-float=soft"
       XPURE64=""
-      XTARGET=$XARCH-linux-musleabi
+      XTARGET=$XARCH-$TVEND-linux-musleabi
       ;;
     armv5 | armv5te )
       XARCH=armv5te
@@ -250,7 +251,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-arch=armv5te --with-float=soft"
       XPURE64=""
-      XTARGET=$XARCH-linux-musleabi
+      XTARGET=$XARCH-$TVEND-linux-musleabi
       ;;
     armv6 )
       XARCH=armv6
@@ -258,7 +259,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-arch=$XARCH --with-fpu=vfp --with-float=hard"
       XPURE64=""
-      XTARGET=$XARCH-linux-musleabihf
+      XTARGET=$XARCH-$TVEND-linux-musleabihf
       ;;
     arm | armv6kz | armv6zk | bcm2835)
       XARCH=armv6kz
@@ -266,7 +267,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-arch=$XARCH --with-tune=arm1176jzf-s --with-abi=aapcs-linux --with-fpu=vfpv2 --with-float=hard"
       XPURE64=""
-      XTARGET=$XARCH-linux-musleabihf
+      XTARGET=$XARCH-$TVEND-linux-musleabihf
       ;;
     armv7 | armv7-a)
       XARCH=armv7-a
@@ -274,7 +275,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-arch=$XARCH --with-fpu=vfpv3 --with-float=hard"
       XPURE64=""
-      XTARGET=$LARCH-linux-musleabihf
+      XTARGET=$LARCH-$TVEND-linux-musleabihf
       ;;
     i486)
       XARCH=$1
@@ -282,7 +283,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-arch=$1 --with-tune=generic"
       XPURE64=""
-      XTARGET=$1-linux-musl
+      XTARGET=$1-$TVEND-linux-musl
       ;;
     i586)
       XARCH=$1
@@ -290,7 +291,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-arch=$1 --with-tune=generic"
       XPURE64=""
-      XTARGET=$1-linux-musl
+      XTARGET=$1-$TVEND-linux-musl
       ;;
     i386 | i686 | x86)
       XARCH=i686
@@ -298,7 +299,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-arch=$XARCH --with-tune=generic"
       XPURE64=""
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     m68k | 68000 | motorola )
       XARCH=m68k
@@ -306,7 +307,7 @@
       MARCH=$XARCH
       XGCCARGS=""
       XPURE64=""
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     mblaze | microblaze | microblazebe | microblazeeb)
       XARCH=microblaze
@@ -314,7 +315,7 @@
       MARCH=$XARCH
       XGCCARGS="--with-endian=big"
       XPURE64=""
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     microblazeel | microblazele)
       XARCH=microblazeel
@@ -322,7 +323,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-endian=little"
       XPURE64=""
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     mips | mips64 | mips64be | mips64eb | mips64r2)
       XARCH=mips64
@@ -330,7 +331,7 @@
       MARCH=$XARCH
       XGCCARGS="--with-endian=big --with-arch=${XARCH}r2 --with-abi=64 --with-float=hard"
       XPURE64=$XARCH
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     loongson | loongson3 | mips64el | mips64le | mips64elr2 | mips64r2el)
       XARCH=mips64el
@@ -338,7 +339,7 @@
       MARCH=${LARCH}64
       XGCCARGS="--with-endian=little --with-arch=${MARCH}r2 --with-abi=64 --with-float=hard"
       XPURE64=$MARCH
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     mips64r6 | mipsisa64r6)
       XARCH=mipsisa64r6
@@ -346,7 +347,7 @@
       MARCH=${LARCH}64
       XGCCARGS="--with-endian=big --with-arch=${MARCH}r6 --with-abi=64 --with-float=hard --with-nan=2008"
       XPURE64=$MARCH
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     mips64r6el | mips64r6le | mipsisa64r6el)
       XARCH=mipsisa64r6el
@@ -354,7 +355,7 @@
       MARCH=${LARCH}64
       XGCCARGS="--with-endian=little --with-arch=${MARCH}r6 --with-abi=64 --with-float=hard --with-nan=2008"
       XPURE64=$MARCH
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     openrisc | or1k | or1ksim)
       XARCH=or1k
@@ -362,7 +363,7 @@
       MARCH=$XARCH
       XGCCARGS=""
       XPURE64=""
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     pmac32 | powerpc | ppc)
       XARCH=powerpc
@@ -370,7 +371,7 @@
       MARCH=$XARCH
       XGCCARGS="--with-cpu=$XARCH --enable-secureplt --without-long-double-128"
       XPURE64=""
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     powerpcle | powerpcel | ppcle | ppcel )
       XARCH=powerpcle
@@ -378,7 +379,7 @@
       MARCH=powerpc
       XGCCARGS="--with-cpu=$LARCH --enable-secureplt --without-long-double-128 --with-endian=little"
       XPURE64=""
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     g5 | powerpc64 | powerpc64be | powerpc64eb | ppc64 | ppc64be | ppc64eb)
       XARCH=powerpc64
@@ -386,7 +387,7 @@
       MARCH=$XARCH
       XGCCARGS="--with-cpu=$XARCH --with-abi=elfv2"
       XPURE64=$XARCH
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     powernv | powerpc64le | ppc64le)
       XARCH=powerpc64le
@@ -394,7 +395,7 @@
       MARCH=${LARCH}64
       XGCCARGS="--with-cpu=$XARCH --enable-secureplt --without-long-double-128 --with-abi=elfv2"
       XPURE64=$MARCH
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     riscv | riscv64 | rv64gc | sifive)
       XARCH=riscv64
@@ -402,7 +403,7 @@
       MARCH=$XARCH
       XGCCARGS="--with-cpu=sifive-u74 --with-arch=rv64gc --with-tune=sifive-7-series --with-abi=lp64d"
       XPURE64=$XARCH
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     s390 | s390x | z15 | z196)
       XARCH=s390x
@@ -410,7 +411,7 @@
       MARCH=$XARCH
       XGCCARGS="--with-arch=z196 --with-tune=z15 --with-long-double-128"
       XPURE64=$XARCH
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     sh2 | superh | sh2le | sh2el)
       XARCH=sh2
@@ -418,7 +419,7 @@
       MARCH=$LARCH
       XGCCARGS=""
       XPURE64=""
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     sh2-fdpic | superh-fdpic | sh2le-fdpic | sh2el-fdpic)
       XARCH=sh2-fdpic
@@ -426,7 +427,7 @@
       MARCH=$LARCH
       XGCCARGS=""
       XPURE64="--enable-fdpic"
-      XTARGET=sh2-linux-muslfdpic
+      XTARGET=sh2-$TVEND-linux-muslfdpic
       ;;
     sh2be | sh2eb)
       XARCH=sh2eb
@@ -434,7 +435,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-endian=big"
       XPURE64=""
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     sh2be-fdpic | sh2eb-fdpic)
       XARCH=sh2eb-fdpic
@@ -442,7 +443,7 @@
       MARCH=$LARCH
       XGCCARGS="--enable-fdpic --with-endian=big"
       XPURE64=""
-      XTARGET=sh2eb-linux-muslfdpic
+      XTARGET=sh2eb-$TVEND-linux-muslfdpic
       ;;
     sh4 | superh4 | sh4le | sh4el)
       XARCH=sh4
@@ -450,7 +451,7 @@
       MARCH=$LARCH
       XGCCARGS=""
       XPURE64=""
-      XTARGET=$XARCH-linux-musl
+      XTARGET=$XARCH-$TVEND-linux-musl
       ;;
     sh4be-fdpic | sh4eb-fdpic)
       XARCH=sh4eb
@@ -458,7 +459,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-endian=big"
       XPURE64=""
-      XTARGET=$XARCH-linux-muslfdpic
+      XTARGET=$XARCH-$TVEND-linux-muslfdpic
       ;;
     amd64 | x86-64 | x86_64)
       XARCH=x86-64
@@ -466,7 +467,7 @@
       MARCH=$LARCH
       XGCCARGS="--with-arch=$XARCH --with-tune=generic"
       XPURE64=$XARCH
-      XTARGET=$LARCH-linux-musl
+      XTARGET=$LARCH-$TVEND-linux-musl
       ;;
     c | -c | --clean)
       printf "${BLUEC}..${NORMALC} Cleaning mussel...\n"
