--- procps-ng-4.0.4.orig/library/uptime.c       2023-07-11 04:09:18.436786212 -0500
+++ procps-ng-4.0.4/library/uptime.c    2023-12-12 00:13:42.715524125 -0600
@@ -30,7 +30,7 @@
 #include <string.h>
 #include <time.h>
 #include <unistd.h>
-#include <utmp.h>
+#include <utmpx.h>
 #ifdef WITH_SYSTEMD
 #include <systemd/sd-daemon.h>
 #include <systemd/sd-login.h>
@@ -51,19 +51,19 @@
 static int count_users(void)
 {
     int numuser = 0;
-    struct utmp *ut;
+    struct utmpx *u;
 
 #if defined(WITH_SYSTEMD) || defined(WITH_ELOGIND)
     if (sd_booted() > 0)
       return sd_get_sessions(NULL);
 #endif
 
-    setutent();
-    while ((ut = getutent())) {
-    if ((ut->ut_type == USER_PROCESS) && (ut->ut_name[0] != '\0'))
+    setutxent();
+    while ((u = getutxent())) {
+    if ((u->ut_type == USER_PROCESS) && (u->ut_name[0] != '\0'))
         numuser++;
     }
-    endutent();
+    endutxent();
 
     return numuser;
 }
