diff -uNr procps-ng-4.0.2.orig/library/uptime.c procps-ng-4.0.2/library/uptime.c
--- procps-ng-4.0.2.orig/library/uptime.c	2022-10-04 06:34:22.014478454 -0500
+++ procps-ng-4.0.2/library/uptime.c	2023-02-03 19:19:13.673149077 -0600
@@ -29,7 +29,7 @@
 #include <string.h>
 #include <time.h>
 #include <unistd.h>
-#include <utmp.h>
+#include <utmpx.h>
 
 #include "misc.h"
 #include "procps-private.h"
@@ -42,14 +42,14 @@
 static int count_users(void)
 {
     int numuser = 0;
-    struct utmp *ut;
+    struct utmpx *u;
 
-    setutent();
-    while ((ut = getutent())) {
-    if ((ut->ut_type == USER_PROCESS) && (ut->ut_name[0] != '\0'))
+    setutxent();
+    while ((u = getutxent())) {
+    if ((u->ut_type == USER_PROCESS) && (u->ut_name[0] != '\0'))
         numuser++;
     }
-    endutent();
+    endutxent();
 
     return numuser;
 }
