--- systemd-254.orig/src/udev/udevadm-hwdb.c	2023-07-28 03:26:32.000000000 -0500
+++ systemd-254/src/udev/udevadm-hwdb.c	2023-12-10 03:51:07.217836172 -0600
@@ -87,8 +87,6 @@
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                        "Either --update or --test must be used.");
 
-        log_notice("udevadm hwdb is deprecated. Use systemd-hwdb instead.");
-
         if (arg_update) {
                 r = hwdb_update(arg_root, arg_hwdb_bin_dir, arg_strict, true);
                 if (r < 0)
