diff -up binutils-2.37/gas/config/tc-i386.c.12~ binutils-2.37/gas/config/tc-i386.c
--- binutils-2.37/gas/config/tc-i386.c.12~	2021-07-08 13:37:19.000000000 +0200
+++ binutils-2.37/gas/config/tc-i386.c	2021-07-20 20:16:16.148889908 +0200
@@ -34,6 +34,15 @@
 #include "opcodes/i386-init.h"
 #include <limits.h>
 
+#if defined(__linux__) && defined(__x86_64__)
+#include <sys/syscall.h>
+#include <sys/personality.h>
+
+#define is_linux32() ((syscall(SYS_personality, 0xffffffff) & PER_MASK) == PER_LINUX32)
+#else
+#define is_linux32() (0)
+#endif
+
 #ifndef INFER_ADDR_PREFIX
 #define INFER_ADDR_PREFIX 1
 #endif
@@ -2681,7 +2690,7 @@ update_code_flag (int value, int check)
       else
 	as_error = as_bad;
       (*as_error) (_("64bit mode not supported on `%s'."),
-		   cpu_arch_name ? cpu_arch_name : default_arch);
+		   cpu_arch_name ? cpu_arch_name : (is_linux32 () ? "i386" : default_arch));
     }
   if (value == CODE_32BIT && !cpu_arch_flags.bitfield.cpui386)
     {
@@ -2690,7 +2699,7 @@ update_code_flag (int value, int check)
       else
 	as_error = as_bad;
       (*as_error) (_("32bit mode not supported on `%s'."),
-		   cpu_arch_name ? cpu_arch_name : default_arch);
+		   cpu_arch_name ? cpu_arch_name : (is_linux32 () ? "i386" : default_arch));
     }
   stackop_size = '\0';
 }
@@ -2818,7 +2827,7 @@ check_cpu_arch_compatible (const char *n
 	 use default_arch.  */
       arch = cpu_arch_name;
       if (!arch)
-	arch = default_arch;
+	arch = (is_linux32() ? "i386" : default_arch);
     }
 
   /* If we are targeting Intel MCU, we must enable it.  */
@@ -5509,7 +5518,7 @@ parse_insn (char *line, char *mnemonic)
   else
     as_bad (_("`%s' is not supported on `%s%s'"),
 	    current_templates->start->name,
-	    cpu_arch_name ? cpu_arch_name : default_arch,
+	    cpu_arch_name ? cpu_arch_name : ((is_linux32 () ? "i386" : default_arch)),
 	    cpu_sub_arch_name ? cpu_sub_arch_name : "");
 
   return NULL;
@@ -13859,7 +13868,7 @@ md_show_usage (FILE *stream)
 const char *
 i386_target_format (void)
 {
-  if (startswith (default_arch, "x86_64"))
+  if (startswith (default_arch, "x86_64") && !is_linux32())
     {
       update_code_flag (CODE_64BIT, 1);
       if (default_arch[6] == '\0')
@@ -13867,7 +13876,7 @@ i386_target_format (void)
       else
 	x86_elf_abi = X86_64_X32_ABI;
     }
-  else if (!strcmp (default_arch, "i386"))
+  else if (!strcmp (default_arch, "i386") || is_linux32())
     update_code_flag (CODE_32BIT, 1);
   else if (!strcmp (default_arch, "iamcu"))
     {
