diff -up binutils-2.37.90/ld/ldelf.c.14~ binutils-2.37.90/ld/ldelf.c
--- binutils-2.37.90/ld/ldelf.c.14~	2022-02-03 11:55:23.709625422 +0100
+++ binutils-2.37.90/ld/ldelf.c	2022-02-03 11:55:23.731625518 +0100
@@ -47,7 +47,7 @@ struct dt_needed
 };
 
 /* Style of .note.gnu.build-id section.  */
-const char *ldelf_emit_note_gnu_build_id;
+const char *ldelf_emit_note_gnu_build_id = NULL;
 
 /* These variables are required to pass information back and forth
    between after_open and check_needed and stat_needed and vercheck.  */
@@ -719,6 +719,9 @@ ldelf_check_ld_elf_hints (const struct b
       FILE *f;
       char *tmppath;
 
+      if (!ldelf_emit_note_gnu_build_id)
+        ldelf_emit_note_gnu_build_id = xstrdup("sha1");
+
       tmppath = concat (ld_sysroot, _PATH_ELF_HINTS, (const char *) NULL);
       f = fopen (tmppath, FOPEN_RB);
       free (tmppath);
diff -up binutils-2.37.90/ld/ldmain.c.14~ binutils-2.37.90/ld/ldmain.c
--- binutils-2.37.90/ld/ldmain.c.14~	2022-02-03 11:55:23.731625518 +0100
+++ binutils-2.37.90/ld/ldmain.c	2022-02-03 11:56:37.471946630 +0100
@@ -331,8 +331,11 @@ main (int argc, char **argv)
 
   link_info.allow_undefined_version = true;
   link_info.keep_memory = true;
+  link_info.optimize = true;
+  link_info.new_dtags = true;
   link_info.max_cache_size = (bfd_size_type) -1;
   link_info.combreloc = true;
+  link_info.relro = true;
   link_info.strip_discarded = true;
   link_info.prohibit_multiple_definition_absolute = false;
   link_info.textrel_check = DEFAULT_LD_TEXTREL_CHECK;
