--- llvm-project-15.0.6.src.orig/llvm/utils/gn/secondary/libcxx/include/BUILD.gn	2022-11-29 04:05:58.000000000 -0600
+++ llvm-project-15.0.6.src/llvm/utils/gn/secondary/libcxx/include/BUILD.gn	2023-01-20 10:33:12.271136647 -0600
@@ -554,7 +554,6 @@
       "__support/ibm/locale_mgmt_zos.h",
       "__support/ibm/nanosleep.h",
       "__support/ibm/xlocale.h",
-      "__support/musl/xlocale.h",
       "__support/newlib/xlocale.h",
       "__support/openbsd/xlocale.h",
       "__support/solaris/floatingpoint.h",
--- llvm-project-15.0.6.src.orig/libcxx/include/__locale	2022-11-29 04:05:58.000000000 -0600
+++ llvm-project-15.0.6.src/libcxx/include/__locale	2023-01-20 10:36:26.683125856 -0600
@@ -37,11 +37,11 @@
 # include <xlocale.h>
 #elif defined(__Fuchsia__)
 # include <__support/fuchsia/xlocale.h>
-#elif defined(__wasi__)
+//#elif defined(__wasi__)
 // WASI libc uses musl's locales support.
-# include <__support/musl/xlocale.h>
-#elif defined(_LIBCPP_HAS_MUSL_LIBC)
-# include <__support/musl/xlocale.h>
+//# include <__support/musl/xlocale.h>
+//#elif defined(_LIBCPP_HAS_MUSL_LIBC)
+//# include <__support/musl/xlocale.h>
 #endif
 
 #if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
--- llvm-project-15.0.6.src.orig/libcxx/include/CMakeLists.txt	2022-11-29 04:05:58.000000000 -0600
+++ llvm-project-15.0.6.src/libcxx/include/CMakeLists.txt	2023-01-20 10:38:42.740118304 -0600
@@ -492,7 +492,6 @@
   __support/ibm/locale_mgmt_zos.h
   __support/ibm/nanosleep.h
   __support/ibm/xlocale.h
-  __support/musl/xlocale.h
   __support/newlib/xlocale.h
   __support/openbsd/xlocale.h
   __support/solaris/floatingpoint.h
